src/compiler/diagnosticMessages.json

```diff
@@ -1546,6 +1546,52 @@
         "code": 2510
     },
 
+    "JSX element attributes type '{0}' must be an object type.": {
+        "category": "Error",
+        "code": 2600
+    },
+    "The return type of a JSX element constructor must return an object type.": {
+        "category": "Error",
+        "code": 2601
+    },
+    "The global type 'JSX.Element' must exist when using JSX.": {
+        "category": "Error",
+        "code": 2602
+    },
+    "Property '{0}' in type '{1}' is not assignable to type '{2}'": {
+        "category": "Error",
+        "code": 2603
+    },
+    "JSX element '{0}' is not a constructor function.": {
+        "category": "Error",
+        "code": 2604
+    },
+    "JSX element '{0}' is not a constructor function for JSX elements.": {
+        "category": "Error",
+        "code": 2605
+    },
+    "Property '{0}' of JSX spread attribute is not assignable to target property.": {
+        "category": "Error",
+        "code": 2606
+    },
+    "JSX element class does not support attributes because it does not have a '{0}' property": {
+        "category": "Error",
+        "code": 2607
+    },
+    "The global type 'JSX.{0}' may not have more than one property": {
+        "category": "Error",
+        "code": 2608
+    },
+    "JSX element implicitly has type 'any' because no interface JSX.{0} exists": {

```

Comment:
Put single quotes around the `JSX.{0}`.

Should this be by the `noImplicitAny` flags?